src/models/keyframesmodel.cpp

```diff
@@ -583,20 +581,102 @@ bool KeyframesModel::isKeyframe(int parameterIndex, int position)
     return false;
 }
 
+bool KeyframesModel::advancedKeyframesInUse()
+{
+    if (m_filter && m_metadata && m_filter->animateIn() <= 0 && m_filter->animateOut() <= 0)
+    for (int i = 0; i < m_metadata->keyframes()->parameterCount(); i++) {
+        if (m_filter->keyframeCount(m_metadata->keyframes()->parameter(i)->property()) > 0) {
+            return true;
+        }
+    }
+    return false;
+}
+
+void KeyframesModel::removeAdvancedKeyframes()
+{
+    if (m_filter && m_metadata && m_filter->animateIn() <= 0 && m_filter->animateOut() <= 0) {
+        for (int i = 0; i < m_metadata->keyframes()->parameterCount(); i++) {
+            QString name = m_metadata->keyframes()->parameter(i)->property();
+            if (m_filter->keyframeCount(name) > 0) {
+                m_filter->set(name, m_filter->get(name, 0));
+                for (auto& key : m_metadata->keyframes()->parameter(i)->gangedProperties()) {
+                    m_filter->set(key, m_filter->get(key, 0));
+                }
+            }
+        }
+        reload();
+    }
+}
+
+bool KeyframesModel::simpleKeyframesInUse()
+{
+    if (m_filter && m_metadata && (m_filter->animateIn() > 0 || m_filter->animateOut() > 0))
+        return true;
+    return false;
+}
+
+void KeyframesModel::removeSimpleKeyframes()
+{
+    if (simpleKeyframesInUse()) {
```

Comment:
`removeSimpleKeyframes()` is never called! And this is the only call `simpleKeyframesInUse()`. Both can be removed. If you want to keep them for completeness, then it should take an argument for the current frame position and get the values at that position.

Reply:
It is called in QML. When the user has simple keyframes in use, and they click the advanced keyframes "clock" button, they are prompted to ask if they want to convert simple keyframes to advanced keyframes. This function is called if they choose "yes".

I was debating if this function should be in KeyframesModel or QmlFilter. I decided on Keyframes model since it has some other similar convenience functions. But I am open to suggestions.

Reply:
Sorry, now I found it. I was searching the PR web page but had a file makred as viewed, which was hiding it.